src/main/groovy/org/cadixdev/gradle/licenser/LicenseExtension.groovy

```diff
@@ -103,16 +104,39 @@ class LicenseExtension extends LicenseProperties {
     @Incubating
     final NamedDomainObjectContainer<LicenseTaskProperties> tasks
 
-    LicenseExtension(Project project) {
-        super(new PatternSet())
+    @PackageScope final ObjectFactory objects
+    @PackageScope final TextResourceFactory textResources
+    @PackageScope final ProviderFactory providers
 
-        this.tasks = project.container(LicenseTaskProperties) { name ->
-            new LicenseTaskProperties((filter as PatternSet).intersect(), name)
+    @Inject
+    LicenseExtension(final ObjectFactory objects, final Project project) {
+        super(new PatternSet(), objects, project.resources.text)
+
+        this.objects = objects
+        this.textResources = project.resources.text
+        this.providers = project.providers
+        this.tasks = objects.domainObjectContainer(LicenseTaskProperties) { String name ->
+            new LicenseTaskProperties((filter as PatternSet).intersect(), name, objects, textResources, charset)
         }
+        this.conditionalProperties = objects.listProperty(LicenseProperties)
 
         // Defaults
-        newLine = true
-        charset = 'UTF-8'
+        this.keywords = objects.listProperty(String).convention(['Copyright', 'License'])
+        this.ignoreFailures = objects.property(Boolean).convention(false)
+        this.skipExistingHeaders = objects.property(Boolean).convention(false)
+        this.charset = objects.property(String).convention('UTF-8')
+        super.charset.set(this.charset)
```

Comment:
Would it be possible to initialize the `charset` property only in the constructor of `LicenseProperties`?